webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,257 @@
+            org: store.organization,
+            gitlabURL: store.gitlabURL,
+            rhsState: store.rhsState,
+        };
+    });
+
+    const [updatedPrs, setUpdatedPrs] = useState<Item[]>(yourPrs);
+    const [RHSState, setRHSState] = useState('');
+    const [updatedReviews, setUpdatedReviews] = useState<Item[]>(reviews);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        setUpdatedReviews(mapPrsToDetails(reviews, reviewDetails));
+    }, [reviews, reviewDetails]);
+
+    useEffect(() => {
+        setUpdatedPrs(mapPrsToDetails(yourPrs, yourPrDetails));
+    }, [yourPrs, yourPrDetails]);
+
+    useEffect(() => {
+        if (yourPrs && rhsState === RHSStates.PRS) {
+            dispatch(getYourPrDetails(
+                mapGitlabItemListToPrList(yourPrs),
+            ));
+            if (reviews && rhsState === RHSStates.REVIEWS) {
+                dispatch(getReviewDetails(
+                    mapGitlabItemListToPrList(reviews),
+                ));
+            }
+        }
+    }, []);
+
+    useEffect(() => {
+        if (shouldUpdateDetails(yourPrs, updatedPrs, RHSStates.PRS, rhsState, RHSState)) {
+            setUpdatedPrs(yourPrs);
+            setRHSState(rhsState);
+            dispatch(getYourPrDetails(mapGitlabItemListToPrList(yourPrs)));
+        }
+    }, [yourPrs, rhsState]);
+
+    useEffect(() => {
+        if (shouldUpdateDetails(reviews, updatedReviews, RHSStates.REVIEWS, rhsState, RHSState)) {
+            setUpdatedReviews(reviews);
+            setRHSState(rhsState);
+            dispatch(getReviewDetails(mapGitlabItemListToPrList(reviews)));
+        }
+    }, [reviews, rhsState]);
+
+    const style = getStyle(theme);
+    const baseURL = gitlabURL || 'https://gitlab.com';
+    let orgQuery = '/dashboard'; //default == all orgs
+    if (org) {
+        orgQuery = `/groups/${org}/-`;
+    }
+
+    let title = '';
+    let gitlabItems: Item[] = [];
+    let listUrl = '';
+
+    switch (rhsState) {
+    case RHSStates.PRS:
+        gitlabItems = yourPrs;
+        title = 'Your Open Merge Requests';
+        listUrl = `${baseURL}${orgQuery}/merge_requests?state=opened&author_username=${username}`;
+        break;
+    case RHSStates.REVIEWS:
+        gitlabItems = reviews;
+        listUrl = `${baseURL}${orgQuery}/merge_requests?reviewer_username=${username}`;
+        title = 'Merge Requests Needing Review';
+        break;
+    case RHSStates.UNREADS:
+        gitlabItems = unreads;
+        title = 'Unread Messages';
+        listUrl = `${baseURL}/dashboard/todos`;
+        break;
+    case RHSStates.ASSIGNMENTS:
+        gitlabItems = yourAssignments;
+        title = 'Your Assignments';
+        listUrl = `${baseURL}${orgQuery}/issues?assignee_username=${username}`;
+        break;
+    default:
+        break;
+    }
+
+    let renderedGitlabItems: React.ReactNode = <div style={style.container}>{'You have no active items'}</div>;
+    if (gitlabItems?.length) {
```

Comment:
Use ternary operators here.

Reply:
previously there was a ternary operator, a comment was present in the PR to do like this

Reply:
Can you paste the link for that comment here?

Reply:
https://github.com/mattermost/mattermost-plugin-gitlab/pull/300#discussion_r920510264